test/parallel/test-runner-watch-mode.mjs

```diff
@@ -48,21 +48,62 @@ async function testWatch({ fileToUpdate, file }) {
     if (testRuns?.length >= 2) ran2.resolve();
   });
 
-  await ran1.promise;
-  runs.push(currentRun);
-  currentRun = '';
-  const content = fixtureContent[fileToUpdate];
-  const path = fixturePaths[fileToUpdate];
-  const interval = setInterval(() => writeFileSync(path, content), common.platformTimeout(1000));
-  await ran2.promise;
-  runs.push(currentRun);
-  clearInterval(interval);
-  child.kill();
-  for (const run of runs) {
-    assert.match(run, /# tests 1/);
-    assert.match(run, /# pass 1/);
-    assert.match(run, /# fail 0/);
-    assert.match(run, /# cancelled 0/);
+  if (action === 'update') {
```

Comment:
can you extract these ifs into three functions (update/rename/delete)?

Reply:
at least the action performed between two runs?

Reply:
fixed, could you take another look 👀